client-state-machine.go

```diff
@@ -113,7 +131,7 @@ func (state clientStateStart) Next(hr handshakeMessageReader) (HandshakeState, [
 		return nil, nil, AlertInternalError
 	}
 	for _, ext := range []ExtensionBody{&sv, &sni, &ks, &sg, &sa} {
-		err := ch.Extensions.Add(ext)
+		err = ch.Extensions.Add(ext)
```

Comment:
Why is this change needed? You seem to shadow it later everywhere relevant.

Reply:
Reverted.